odonto/odonto_submissions/models.py

```diff
@@ -134,21 +134,26 @@ def get_rejected_submissions(self):
         if "respce" not in self.content_as_dict["icset"]["ic"]:
             return {}
 
-        responses = self.content_as_dict["icset"]["ic"]["respce"]
-        if not isinstance(responses, list):
-            responses = [responses]
-        for i in responses:
-            response = i["rsp"]
-            submission = self.get_all_submissions().filter(
-                claim__reference_number=int(response["@clrn"])
-            ).get()
-
-            if isinstance(response["mstxt"], list):
-                result[submission] = ", ".join([
-                    y["#text"] for y in response["mstxt"]
-                ])
-            else:
-                result[submission] = response["mstxt"]["#text"]
+        responses_sections = self.content_as_dict["icset"]["ic"]["respce"]
+        if not isinstance(responses_sections, list):
+            responses_sections = [responses_sections]
+        for response_section in responses_sections:
+            responses = response_section["rsp"]
+
+            if not isinstance(responses, list):
```

Comment:
It's like you can't accurately represent XML as a python dictionary without the XSD to know whether elements are potentially repeatable ;)